Content/NPCs/Town/Hunter.cs

```diff
@@ -0,0 +1,179 @@
+using System.Collections.Generic;
+using PathOfTerraria.Content.Items.Gear.Weapons.Bow;
+using PathOfTerraria.Core.Systems.Questing;
+using PathOfTerraria.Helpers.Extensions;
+using ReLogic.Content;
+using Terraria.GameContent;
+using Terraria.ID;
+using Terraria.Localization;
+
+namespace PathOfTerraria.Content.NPCs.Town;
+
+[AutoloadHead]
+public class Hunter : ModNPC
+{
+	/// <summary>
+	///     The index of the current dialogue sentence.
+	/// </summary>
+	public int Dialogue
+	{
+		get => dialogue;
+		set => dialogue = value > 2 ? 0 : value;
+	}
+
+	public override string HeadTexture { get; } = $"{nameof(PathOfTerraria)}/Assets/NPCs/Town/Hunter_Head";
+
+	public override string Texture { get; } = $"{nameof(PathOfTerraria)}/Assets/NPCs/Town/Hunter";
+
+	private int dialogue;
+
+	public override void SetStaticDefaults()
+	{
+		Main.npcFrameCount[Type] = 23;
+
+		NPCID.Sets.NoTownNPCHappiness[Type] = true;
+
+		NPCID.Sets.DangerDetectRange[Type] = 400;
+
+		NPCID.Sets.AttackFrameCount[Type] = 4;
+		NPCID.Sets.AttackType[Type] = 1;
+		NPCID.Sets.AttackTime[Type] = 60;
+		NPCID.Sets.AttackAverageChance[Type] = 10;
+
+		NPCID.Sets.NPCBestiaryDrawModifiers modifiers = new() { Velocity = 1f };
+
+		NPCID.Sets.NPCBestiaryDrawOffset.Add(Type, modifiers);
+	}
+
+	public override void SetDefaults()
+	{
+		NPC.townNPC = true;
+		NPC.friendly = true;
+
+		NPC.lifeMax = 250;
+
+		NPC.width = 30;
+		NPC.height = 50;
+
+		NPC.knockBackResist = 0.4f;
+
+		NPC.HitSound = SoundID.NPCHit1;
+		NPC.DeathSound = SoundID.NPCDeath1;
+
+		NPC.aiStyle = NPCAIStyleID.Passive;
+
+		AnimationType = NPCID.BestiaryGirl;
+	}
+
+	public override ITownNPCProfile TownNPCProfile()
+	{
+		return this.GetDefaultTownProfile();
+	}
+
+	public override List<string> SetNPCNameList()
+	{
+		return [DisplayName.Value];
```

Comment:
I believe one should return `[]` or `null` (an empty collection is preferable) to avoid this. https://github.com/tModLoader/tModLoader/blob/3dc99738621ee560eb8c819d8625a77f3b4630c6/patches/tModLoader/Terraria/NPC.cs.patch#L661 tModLoader does not introduce any special behavior to handling an empty string from what I can tell, and while vanilla may handle it, this also makes our intentions clearer.